### PR TITLE
Add config option to more aggressively allow overflow

### PR DESCRIPTION
**Pull request by kestred**
Not sure if `overflow_delimited_expr` is the best name for the config option, so recommendations very much appreciated.  Very happy to fix/change anything in the PR as necessary.

I presume this is common for most folks adding a config option, but I'd obvi- love if this could be enabled by default 😊 .

### CODE REVIEW

**Comment by otavio** [comment]
I like the end result, it looks beautiful and clean. I did not review if the style guide has any explicit reference for it though.

File: tests/source/expr-overflow-delimited.rs [hunk 1]
```diff
@@ -0,0 +1,111 @@
+// rustfmt-overflow_delimited_expr: true
```
Comment (nrc) [inline tests/source/expr-overflow-delimited.rs]: Could you add a test with a comment before the overflowing argument please?

File: src/expr.rs [hunk 2]
```diff
@@ -1343,12 +1343,31 @@ pub fn can_be_overflowed_expr(context: &RewriteContext, expr: &ast::Expr, args_l
...
-        | ast::ExprKind::Call(..)
-        | ast::ExprKind::Mac(..)
-        | ast::ExprKind::MethodCall(..)
-        | ast::ExprKind::Struct(..)
-        | ast::ExprKind::Tup(..) => context.use_block_indent() && args_len == 1,
+
+        // Handle `[]` and `{}`-like expressions
+        ast::ExprKind::Array(..) | ast::ExprKind::Struct(..) => {
+            if context.config.overflow_delimited_expr() {
+                context.use_block_indent() || context.config.indent_style() == IndentStyle::Visual
```
Comment (nrc) [inline src/expr.rs]: This conditional seems a bit strange - it looks to be always true?

File: tests/source/expr-overflow-delimited.rs (hunk 1 above)
Comment (kestred) [reply tests/source/expr-overflow-delimited.rs]: Boop!

**Pull request merged**